src/overlays/actors/ovl_En_Elfgrp/z_en_elfgrp.c

```diff
@@ -27,54 +45,541 @@ const ActorInit En_Elfgrp_InitVars = {
+                this->actionFunc = func_80A3A520;
+                if ((this->actor.home.rot.z != 0) && Flags_GetSwitch(globalCtx, this->actor.home.rot.z)) {
+                    this->actionFunc = func_80A3A600;
+                } else if (INV_CONTENT(ITEM_MASK_GREAT_FAIRY) == ITEM_MASK_GREAT_FAIRY) {
+                    func_80A396B0(this, 4);
+                } else if (INV_CONTENT(ITEM_MASK_DEKU) != ITEM_MASK_DEKU) {
+                    func_80A396B0(this, 5);
+                } else {
+                    this->unk_14A |= 2;
+                    func_80A396B0(this, 6);
+                }
+            } else if ((gSaveContext.weekEventReg[8] & 0x80)) {
+                func_80A39DC8(this, globalCtx, 24, 0);
+                this->actionFunc = func_80A3A398;
+                if (INV_CONTENT(ITEM_MASK_DEKU) == ITEM_MASK_DEKU) {
+                    if (INV_CONTENT(ITEM_MASK_GREAT_FAIRY) == ITEM_MASK_GREAT_FAIRY) {
+                        func_80A396B0(this, 2);
+                    } else {
+                        func_80A396B0(this, 3);
+                        this->unk_14A |= 2;
+                    }
+                } else if (gSaveContext.magicAcquired == true) {
+                    func_80A396B0(this, 1);
+                }
+            } else {
+                func_80A39DC8(this, globalCtx, 24, 0);
+                this->actionFunc = func_80A3A8F8;
+                if ((gSaveContext.weekEventReg[9] & this->unk_146)) {
+                    this->actor.textId = 0x580;
+                } else {
+                    this->actor.textId = 0x578;
+                }
+                this->actor.flags |= (ACTOR_FLAG_1 | ACTOR_FLAG_8);
+            }
+            break;
+    }
+}
+
+void EnElfgrp_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+s32 func_80A39BD0(GlobalContext* globalCtx, s32 arg2) {
+    s8 sp1F;
+
+    if ((arg2 < 1) || (arg2 >= 5)) {
+        return 0;
+    }
+
+    sp1F = gSaveContext.inventory.strayFairies[arg2 - 1];
+    return (sp1F - func_80A39C1C(globalCtx, arg2)) + 10;
+}
+
+s32 func_80A39C1C(GlobalContext* globalCtx, s32 arg1) {
+    // the permanentSceneFlags access here is in the form
+    // struct {
+    //   u32 clockTown : 1;
+    //   u32 fountains[4] : 5;
+    // } FairyFountains;
+    // where arg1 is:
+    //    0: clocktown
+    //    1: woodfall,
+    //    2: snowhead,
+    //    3: great bay,
+    //    4: stone tower
+    // clocktown is handled separately, and then the fountains are looked up as array indexing
+
+    s32 temp_v1;
+
+    if ((arg1 < 0) || (arg1 >= 5)) {
+        return 0;
+    }
+
+    if (arg1 == 0) {
+        if (gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 & 1) {
+            return 25;
+        }
+        return 24;
+    }
+
+    temp_v1 = (gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 >> (((arg1 - 1) * 5) + 1)) & 0x1F;
+    if (temp_v1 < 10) {
+        temp_v1 = 10;
+    } else if (temp_v1 > 25) {
+        temp_v1 = 25;
+    }
```

Comment:
Could this be made into a clamp?

Reply:
Nope, the redundant `temp_v1 = temp_v1` part isn't removed.